source/AI.cpp

```diff
@@ -3723,7 +3742,8 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 			int index = 0;
 			for(const Hardpoint &hardpoint : ship.Weapons())
 			{
-				if(hardpoint.IsReady() && !hardpoint.GetOutfit()->Icon())
+				if(hardpoint.IsReady() && !hardpoint.GetOutfit()->Icon()
+						&& (!shift || TargetInRange(ship, hardpoint)))
```

Comment:
Booleans are beyond me and it doesnt seem right to me but I assume you've tested it and it works?

Reply:
It can be stated as

“Either shift is not held, or you are in range.”

Reply:
But looking at the title I thought it was shift is held means we only fire at targets in range, not the opposite?